airflow/ui/src/components/TriggerDag/TriggerDAGModal.tsx

```diff
@@ -41,33 +57,102 @@ const TriggerDAGModal: React.FC<TriggerDAGModalProps> = ({
   onClose,
   open,
 }) => {
-  const initialDagParams = useMemo(
-    () => ({
-      configJson: "{}",
-      dagId,
-      dataIntervalEnd: "",
-      dataIntervalStart: "",
-      notes: "",
-      runId: "",
-    }),
-    [dagId],
-  );
+  const { data, error } = useDagServiceGetDagDetails({ dagId });
 
-  const [dagParams, setDagParams] = useState<DagParams>(initialDagParams);
+  const transformedParams = data?.params
+    ? Object.fromEntries(
+        Object.entries(data.params).map(([key, param]) => [
+          key,
+          (param as { value: unknown }).value,
+        ]),
+      )
+    : {};
+
+  const initialConf = JSON.stringify(transformedParams, undefined, 2);
+
+  const [dagParams, setDagParams] = useState<DagParams>({
+    configJson: initialConf,
+    dagId,
+    dataIntervalEnd: "",
+    dataIntervalStart: "",
+    notes: "",
+    runId: "",
+  });
+
+  useEffect(() => {
+    const newConfigJson = Boolean(error) ? "{}" : initialConf; // Fallback to "{}" if there's an error
 
-  const handleTrigger = useCallback(
+    setDagParams((prevParams) => ({
+      ...prevParams,
+      configJson: newConfigJson,
+    }));
+  }, [initialConf, error]);
+
+  const queryClient = useQueryClient();
+  const onSuccess = async () => {
+    await queryClient.invalidateQueries({
+      queryKey: [useDagServiceGetDagsKey],
+    });
+
+    await queryClient.invalidateQueries({
+      queryKey: [useDagsServiceRecentDagRunsKey],
+    });
+
+    await queryClient.invalidateQueries({
+      queryKey: [useDagRunServiceGetDagRunsKey],
+    });
+  };
+  const { mutate } = useDagRunServiceTriggerDagRun({
+    onSuccess,
+  });
+
+  const onTrigger = useCallback(
```

Comment:
I think we should probably move a lot of the mutation logic into a custom mutation under `/queries/useTrigger.ts` and then the component only worries about editing the form and not its side effects

Reply:
Actual yes, we should make this component light.